packages/kit/src/runtime/server/resolve.js

```diff
@@ -0,0 +1,171 @@
+import { DEV } from 'esm-env';
+import { validate_server_exports } from '../../utils/exports.js';
+import { exec } from '../../utils/routing.js';
+import { decode_pathname, decode_params } from '../../utils/url.js';
+import { base } from '__sveltekit/paths';
+
+/** @type {Partial<import('crossws').Hooks>} */
+const noHooks = {};
+
+/**
+ * @param {import('types').SSROptions} options
+ * @param {import('@sveltejs/kit').SSRManifest} manifest
+ * @param {import('types').SSRState} state
+ * @returns {(info: RequestInit | import('crossws').Peer) => Promise<Partial<import('crossws').Hooks>>}
+ */
+export function resolve(options, manifest, state) {
+	return async (info) => {
+		/** @type {RequestInit} */
+		let request;
+
+		// Check if info is a Peer object
+		if ('request' in info) {
+			request = info.request;
+		} else {
+			request = info;
+		}
+
+		/** URL but stripped from the potential `/__data.json` suffix and its search param  */
+		// @ts-expect-error URL is actually valid on this type, I think due to the NodeReqProxy used.
+		const url = new URL(request.url);
+
+		// reroute could alter the given URL, so we pass a copy
+		let rerouted_path;
+		try {
+			rerouted_path = options.hooks.reroute({ url }) ?? url.pathname;
+		} catch {
+			return noHooks;
+		}
+
+		let decoded;
+		try {
+			decoded = decode_pathname(rerouted_path);
+		} catch (e) {
+			console.error(e);
+
+			return noHooks;
+		}
+
+		if (base && decoded.startsWith(base)) {
+			decoded = decoded.slice(base.length) || '/';
+		}
```

Comment:
If this the code block below is the parallel...
https://github.com/sveltejs/kit/blob/461aa9584ee7a40e3c25a66be4768e98ec16cdd8/packages/kit/src/runtime/server/respond.js#L137-L142
...shouldn't something like the below be the equivalent? So that it ignores paths not starting with the chosen `base`
```suggestion
		if (base && !state.prerendering?.fallback) {
			if (!decoded.startsWith(base)) {
				return noHooks;
			}
			decoded = decoded.slice(base.length) || '/';
		}
```

Reply:
Yes I think this is a good change. Thank you @eltigerchino!